application/backend/src/main/resources/apps/typerefinery/components/widgets/chart/clientlibs/functions.js

```diff
@@ -0,0 +1,48 @@
+window.Typerefinery = window.Typerefinery || {};
+window.Typerefinery.Components = Typerefinery.Components || {};
+window.Typerefinery.Components.Widgets = Typerefinery.Components.Widgets || {};
+window.Typerefinery.Components.Widgets.Charts = Typerefinery.Components.Widgets.Charts || {};
+window.Typerefinery.Components.Widgets.Charts.Variants = Typerefinery.Components.Widgets.Charts.Variants || {};
+
+(function (ns, componentNs, variantNs, document, window) {
+  "use strict";
+
+  ns.renderChart = ($component) => {
+    
+  }
+
+  ns.dataReceived = (data, $component) => {
+    // parse json value from data-model attribute as component config.
+    const componentConfig = componentNs.getComponentConfig($component);
+   
+    // switch case for chart variants.
+    switch (componentConfig.chartVariant) {
+      case "barChart":
+        variantNs?.BarChart?.dataReceived(data, $component);
+        break;
+      case "pieChart":
+        variantNs?.PieChart?.dataReceived(data, $component);
+        break;
+      default:
+        variantNs?.LineChart?.dataReceived(data, $component);
+    }
+  };
+
+  ns.init = ($component) => { 
+    // parse json value from data-model attribute as component config.
+    const componentConfig = componentNs.getComponentConfig($component);
+   
+    // switch case for chart variants.
+    switch (componentConfig.chartVariant) {
+      case "barChart":
+        variantNs?.BarChart?.init($component);
```

Comment:
this logic should be in behaviour, no one will know that you are loading another init hidden within init. this should be simple.

Reply:
Ok, will update this.